### PR TITLE
Fix parsing Z, M, and ZM WKT strings

### PR DESCRIPTION
**Pull request by kylebarron**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
- [x] I ran cargo fmt
---

### CODE REVIEW

**Comment by kylebarron** [comment]
@michaelkirk since the changes from https://github.com/georust/wkt/pull/110 might be fresh on your mind, do you want to look at the merge conflicts here and/or do you have suggestions for how to resolve them? Or do you want to look at the approach I took here and maybe rebase those onto the latest main?

**Comment by kylebarron** [comment]
Some of the tests here come from the GEOS test suite in https://github.com/libgeos/geos/blob/8e525c825caf314d61a6279a426a59ded115681e/tests/unit/io/WKTReaderTest.cpp

**Comment by michaelkirk** [comment]
> @michaelkirk since the changes from https://github.com/georust/wkt/pull/110 might be fresh on your mind, do you want to look at the merge conflicts here and/or do you have suggestions for how to resolve them? Or do you want to look at the approach I took here and maybe rebase those onto the latest main?

Sorry, I wouldn't have merged that if I knew you were working on this. I'll push up a resolution in a moment the changes are pretty mechanical.

**Comment by michaelkirk** [comment]
I'm seeing some tests failing with your branch. Can you fix those up?

```
failures:

---- tests::empty_items stdout ----
thread 'tests::empty_items' panicked at src/lib.rs:380:63:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::lowercase_point stdout ----
thread 'tests::lowercase_point' panicked at src/lib.rs:395:63:
called `Option::unwrap()` on a `None` value

---- types::multipoint::tests::empty_multipoint stdout ----
thread 'types::multipoint::tests::empty_multipoint' panicked at src/types/multipoint.rs:151:63:
called `Result::unwrap()` on an `Err` value: "Unexpected word before open paren"


failures:
    tests::empty_items
    tests::lowercase_point
```

**Comment by kylebarron** [comment]
> Sorry, I wouldn't have merged that if I knew you were working on this.

No worries! It was just bad luck!

**Comment by kylebarron** [comment]
tests all pass locally now!

**Comment by kylebarron** [comment]
Do you know why CI isn't running on this branch?

**Comment by michaelkirk** [comment]
I do not know why CI isn't running for your branch. Maybe related to the conflicts in the current branch? (working on those now)

**Comment by michaelkirk** [comment]
Ok - conflicts resolved.

File: src/lib.rs [hunk 1]
```diff
@@ -272,21 +243,65 @@ where
...
 }
 
+fn infer_geom_dimension<T: WktNum + FromStr + Default>(
+    tokens: &mut PeekableTokens<T>,
+) -> Result<Dimension, &'static str> {
+    if let Some(Ok(c)) = tokens.peek() {
+        match c {
+            // If we match a word check if it's Z/M/ZM and consume the token from the stream
+            Token::Word(x) => match x.as_str() {
+                "Z" => {
```
Comment (michaelkirk) [inline src/lib.rs:253]: These should be case insensitive - see our matching elsewhere using `eq_ignore_ascii_case`

File: src/types/mod.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 pub use self::coord::Coord;
+pub use self::dimension::Dimension;
```
Comment (michaelkirk) [inline src/types/mod.rs:16]: Does this need to be public? It doesn't look like it's used in any public APIs.

**Review by michaelkirk — CHANGES_REQUESTED** [review]

**Comment by ariesdevil** [comment]
So we now only support `POINT Z`, and no longer support `POINTZ`?

**Comment by michaelkirk** [comment]
That's correct. I don't ever use 3d points, but my understanding is that "POINT Z" is the correct spelling. Please let us know if you have compelling counter examples!

**Review by michaelkirk — APPROVED** [review]
Nice work, thanks @kylebarron!

**Comment by kylebarron** [comment]
> So we now only support `POINT Z`, and no longer support `POINTZ`?

I added support for `POINTZ` (and `LINESTRINGZ` etc) in the latest commit. It does make the `match` statement kinda verbose; @michaelkirk what do you think?

**Comment by michaelkirk** [comment]
Sure, since there is apparently some precedent for this spelling, I'm OK with (happy to?) support it. Thanks @kylebarron

**Comment by kylebarron** [comment]
I had missed it but chapter 7 of https://www.ogc.org/standard/sfa/ does define WKT for geometries

**Review by michaelkirk — APPROVED** [review]

**Pull request merged**